lgc/builder/MiscBuilder.cpp

```diff
@@ -141,6 +141,10 @@ Instruction *MiscBuilder::CreateEmitMeshTasks(Value *groupCountX, Value *groupCo
 // @returns Instruction to set the actual size of mesh outputs
 Instruction *MiscBuilder::CreateSetMeshOutputs(Value *vertexCount, Value *primitiveCount, const Twine &instName) {
   assert(m_shaderStage == ShaderStageMesh); // Only valid for mesh shader
+
+  auto resUsage = getPipelineState()->getShaderResourceUsage(ShaderStageMesh);
+  resUsage->builtInUsage.mesh.setMeshOutputs = true;
+
```

Comment:
I don't think this approach to guarding works. I believe it would be correct to have a shader that looks like:
```
void main() {
  ...
  if (... workgroup uniform condition ...)
    return;

  SetMeshOutputs(..., ...);
  ... write to outputs ...
}
```
In such a shader, SetMeshOutputs is statically used, but still it's possible to reach the shader epilog without it being called.

How about: initialize vertex and primitive count in LDS to ~0. In the epilog, check if the counts are ~0, and if they are, do sendmsg and use vertex count = primitive count = 0. It implies another branch, which is annoying, but perhaps we can improve on that later.

Another thing I just noticed, though it's in code that's already there: `instName` isn't actually used.

Reply:
Thank you. This is really a good case. I will follow your suggestion. Indeed, my original solution is unable to handle such case.